ArchUnitNETTests/Fluent/Syntax/Elements/ObjectsShouldTests.cs

```diff
@@ -0,0 +1,3229 @@
+
+        helper.AddSnapshotHeader("Violations");
+        should = Types().That().Are(helper.BaseClass).Should();
+        should.NotHaveName(helper.BaseClass.Name).AssertOnlyViolations(helper);
+        should.NotHaveName("^Base.*$", true).AssertOnlyViolations(helper);
+        should.NotHaveFullName(helper.BaseClass.FullName).AssertOnlyViolations(helper);
+        should.NotHaveFullName("^.*\\.Base.*$", true).AssertOnlyViolations(helper);
+        should.NotHaveNameContaining("Base").AssertOnlyViolations(helper);
+        should
+            .NotHaveFullNameContaining(helper.BaseClass.Namespace.Name)
+            .AssertOnlyViolations(helper);
+        should.NotHaveNameStartingWith("Base").AssertOnlyViolations(helper);
+        should.NotHaveNameEndingWith("Class").AssertOnlyViolations(helper);
+        await helper.AssertSnapshotMatches();
+    }
+
+    [Fact]
+    public async Task OnlyDependOnTest()
+    {
+        var helper = new DependencyAssemblyTestHelper();
+        helper.AddSnapshotHeader("No violations");
+        var should = Types().That().Are(helper.ChildClass).Should();
+        should.OnlyDependOn(helper.BaseClass.FullName).AssertNoViolations(helper);
+        should.OnlyDependOn("^.*\\.BaseClass$", true).AssertNoViolations(helper);
+        should.OnlyDependOn([helper.BaseClass.FullName]).AssertNoViolations(helper);
+        should.OnlyDependOn(["^.*\\.BaseClass$"], true).AssertNoViolations(helper);
+        should.OnlyDependOn(helper.BaseClass).AssertNoViolations(helper);
+        should.OnlyDependOn(helper.BaseClassSystemType).AssertNoViolations(helper);
+        should.OnlyDependOn(Classes().That().Are(helper.BaseClass)).AssertNoViolations(helper);
+        should.OnlyDependOn([helper.BaseClass]).AssertNoViolations(helper);
+        should.OnlyDependOn([helper.BaseClassSystemType]).AssertNoViolations(helper);
+
+        helper.AddSnapshotHeader("Violations");
+        should = Types().That().Are(helper.ClassWithMultipleDependencies).Should();
+        should.OnlyDependOn(helper.BaseClass.FullName).AssertOnlyViolations(helper);
+        should.OnlyDependOn([helper.BaseClass.FullName]).AssertOnlyViolations(helper);
+        should.OnlyDependOn(helper.BaseClass).AssertOnlyViolations(helper);
+        should.OnlyDependOn(helper.BaseClassSystemType).AssertOnlyViolations(helper);
+        should.OnlyDependOn(Classes().That().Are(helper.BaseClass)).AssertOnlyViolations(helper);
+        should.OnlyDependOn([helper.BaseClass]).AssertOnlyViolations(helper);
+        should.OnlyDependOn([helper.BaseClassSystemType]).AssertOnlyViolations(helper);
+
+        helper.AddSnapshotHeader("Non-existent type");
+        should = Types().That().Are(helper.ClassWithMultipleDependencies).Should();
+        should.OnlyDependOn(helper.NonExistentObjectName).AssertOnlyViolations(helper);
+        should.OnlyDependOn([helper.NonExistentObjectName]).AssertOnlyViolations(helper);
+
+        helper.AddSnapshotHeader("Type outside of architecture");
```

Comment:
This part seems a bit strange. Is there a reason why we do not check the `ClassWithMultipleDependencies` here like in the tests above or below?